### PR TITLE
Add usage settings for ChangeQuote

### PR DESCRIPTION
**Pull request by lexerom**
## Summary

Add usageSettings to Change quote. It specifies how to handle removed items with usages.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 1]
```diff
@@ -227,6 +227,38 @@ properties:
...
+  usageSettings:
+    type:
+      - 'object'
+      - 'null'
+    description: Settings for items with metered billing.
+    properties:
+      policy:
+        type: string
+        enum:
+          - reset
```
Comment (VladimirRebilly) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:239]: I think users must not be allowed to control this setting. They will always use `reset` to not to pay for the usage

Comment (lexerom) [reply openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:239]: I agree, but this schema is not used for storefront write operations.

**Review by VladimirRebilly — DISMISSED** [review]

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 2]
```diff
@@ -233,6 +233,38 @@ properties:
     description: |-
       Specifies if the quotation invoice must include interim items only.
       If this value is set to `false`, all upcoming items are included.
+  usageSettings:
+    type:
+      - 'object'
+      - 'null'
+    description: Settings for items with metered billing.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:240]: suggestion: to help make it clear that these are usage settings
```suggestion
    description: Usage settings for items with metered billing.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 3]
```diff
@@ -233,6 +233,38 @@ properties:
...
+    properties:
+      policy:
+        type: string
+        enum:
+          - reset
+          - transfer
+        description: Policy for removed items with metered billing plan.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:247]: suggestion: for readability
```suggestion
        description: Policy for removed items that have a metered billing plan.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 4]
```diff
@@ -233,6 +233,38 @@ properties:
...
+        x-enumDescriptions:
+          reset: Delete existing usages.
+          transfer: Transfer usages from removed plans to existing plans inside the subscription with `transfers` field.
+      transfers:
+        type: array
+        default: []
+        description: |-
+          List of usage transfers for items with metered billing plans.
+          If an item with specified plan does not exist or does not have metered billing plan then this transfer is ignored.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:256]: suggestion: for readability
```suggestion
          If an item with a specified plan does not exist, or does not have a metered billing plan, this transfer is ignored.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 5]
```diff
@@ -227,6 +227,38 @@ properties:
     description: |-
       Specifies if the quotation invoice must include interim items only.
       If this value is set to `false`, all upcoming items are included.
+  usageSettings:
+    type:
+      - 'object'
+      - 'null'
+    description: Settings for items with metered billing.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:234]: suggestion: to help make it clear that these are usage settings
```suggestion
    description: Usage settings for items with metered billing.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 6]
```diff
@@ -227,6 +227,38 @@ properties:
...
+    properties:
+      policy:
+        type: string
+        enum:
+          - reset
+          - transfer
+        description: Policy for removed items with metered billing plan.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:241]: suggestion: for readability
```suggestion
        description: Policy for removed items that have a metered billing plan.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 7]
```diff
@@ -227,6 +227,38 @@ properties:
...
+        x-enumDescriptions:
+          reset: Delete existing usages.
+          transfer: Transfer usages from removed plans to existing plans inside the subscription with `transfers` field.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:244]: nitpick: I found this sentence difficult to understand

I've made a suggestion based on my understanding. I'm not sure if it's correct 😅 
```suggestion
          transfer: Transfer usages from removed plans to existing plans that are in the same subscription, using the `transfers` field.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 8]
```diff
@@ -227,6 +227,38 @@ properties:
...
+      transfers:
+        type: array
+        default: []
+        description: |-
+          List of usage transfers for items with metered billing plans.
+          If an item with specified plan does not exist or does not have metered billing plan then this transfer is ignored.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:250]: suggestion: for readability
```suggestion
          If an item with a specified plan does not exist, or does not have a metered billing plan, this transfer is ignored.
```

**Review by adamaltman — DISMISSED** [review]
I agree with @noon-dawg 's suggestions. Looks good though.

**Comment by lexerom** [comment]
> I agree with @noon-dawg 's suggestions. Looks good though.

Yes, I will apply his suggestions. But I have another thought in mind which is "policy per setting":
[
  {planId: 'plan-from', policy: 'transfer', transferTo: 'plan-to'},
  {planId: 'plan2', policy: 'reset'}
]

It works better with multiple metered billing plans (simple example is mobile minutes + gigabytes).

What do you think @adamaltman?

**Comment by adamaltman** [comment]
Looks good.

**Review by noon-dawg — APPROVED** [review]
Looks good on the docs side

**Review by VladimirRebilly — APPROVED** [review]

**Review by lucasxciv — APPROVED** [review]

**Pull request merged**